reader_go18.go

```diff
@@ -149,6 +149,9 @@ func (r *GenericReader[T]) readRows(rows []T) (int, error) {
 		r.base.rowbuf = r.base.rowbuf[:nRequest]
 	}
 
+	schema := r.base.Schema()
+	columns := make([][]Value, len(schema.columns))
```

Comment:
Arguably these columns should be stored in the GenericReader object and reused across Read calls. That would help if Read is called many times with a relatively small rows slice. However, it's harder to know for sure that won't cause problems.

Reply:
I have been thinking about something similar. We use lots os `[]Value` `[][]Value` since  `GenericReader` doesn't guarantee thread safety, we can just reuse them . 

having something like 

```
type ValueBuffer[T Value | []Value] struct {
	Values []T
}

func (v *ValueBuffer[T]) Reserve(n int) {
	v.Values = slices.Grow(v.Values, n)
}

func (v *ValueBuffer[T]) Reset() {
	v.Values = v.Values[:0]
}
```

That can be used internally . This will keep areas we need to change small.  We don't need to tie this to GenericReader or anything, we just pull them from `sync.Pool` use them and release them when we are done.

This guarantee

- Not breaking things
- We can control on allocation strategies on how to grow the buffers

Very simple to reason, since we only focus on where we get `[]Value` or `[][]Value` the rest of the api stays pretty much unchanged.